src/hierarchical-menu/__tests__/hierarchical-menu.spec.ts

```diff
@@ -56,20 +57,41 @@ describe('HierarchicalMenu', () => {
     expect(refine).toHaveBeenCalledWith(defaultState.items[0].value);
   });
 
-  it('should apply `transformItems` if specified', () => {
-    const fixture = render({});
+  it('should create widget with connectHierarchicalMenu and pass instance options', () => {
+    const createWidget = jest.spyOn(
+      NgAisHierarchicalMenu.prototype,
+      'createWidget'
+    );
 
-    const mapItems = items =>
-      items.map(item => ({
-        ...item,
-        label: `foo - ${item.label}`,
-        data: item.data ? mapItems(item.data) : null,
-      }));
+    const transformItems = jest.fn(x => x);
+    const render = createRenderer({
+      defaultState,
+      template: `<ais-hierarchical-menu 
+          [transformItems]="transformItems" 
+          [attributes]="['attr1', 'attr2']" 
+          [limit]="3" 
+          rootPath="attr1" 
+          [showParentLevel]="false"
+          [sortBy]="['count:asc']"
+          separator="/"
+          ></ais-hierarchical-menu>`,
+      TestedWidget: NgAisHierarchicalMenu,
+      additionalDeclarations: [NgAisHierarchicalMenuItem],
+      methods: { transformItems },
```

Comment:
so this is how we can avoid string assertions in other tests? Looks like we need to migrate those

Reply:
(not in this PR of course)

Reply:
yes, we can do it at some point. But since we're mostly just passing a reference to the connector in this layer of the code, the tests using string assertion can be considered enough to secure the feature.